### PR TITLE
Enable file compression

### PR DESCRIPTION
**Pull request by pchico83**
Signed-off-by: Pablo Chico de Guzman <pchico83@gmail.com>

## Proposed changes
- This can speed up the `up` command 50%, mainly for large folders

### CODE REVIEW

**Review by rberrelleza — COMMENTED** [review]
Most repos are going to have lots of small files rather than a few large ones. Will there be a benefit there, or the compress/uncompress time will eat away the benefits?

We could start with this as okteto.yaml parameter, so it's only enabled on repos with big files.

**Comment by pchico83** [comment]
@rberrelleza yes, I was thinking about that too, good idea.
Should we go with `compression: true/false`?

**Comment by rberrelleza** [comment]
> @rberrelleza yes, I was thinking about that too, good idea.
> Should we go with `compression: true/false`?

yeah, but can we put it inside of `sync`? feel like as a standalone it might be confusing. (not sure if you can do list + keys in yaml though).

**Comment by pchico83** [comment]
@rberrelleza it cannot be inside of `sync`. We can do a new one, `syncProperties` or something like that

**Comment by rberrelleza** [comment]
> @rberrelleza it cannot be inside of `sync`. We can do a new one, `syncProperties` or something like that

`syncSettings` and we can put the reload and a few other things there?

**Comment by rlamana** [comment]
Why can't it be inside `sync`? Grouped in an `options` property under the `sync`?

**Comment by pchico83** [comment]
> Why can't it be inside `sync`? Grouped in an `options` property under the `sync`?

that would look like this:
```
sync:
   - .:/app
   options:
      compression: true
```
which is not valid yaml and would be confusing. We would need to group the folders in a new property, like this:
```
sync:
   folders:
      - .:/app
   options:
      compression: true
```
but we would be complicating the syntax for 99% of cases

**Comment by rlamana** [comment]
Ok, makes sense. Then at least `syncOptions` rather than `syncProperties`? 😬

**Comment by rberrelleza** [comment]
could we support both shapes @pchico83 ? (similar to how the build attribute takes a string, or a dict?)

**Comment by pchico83** [comment]
> could we support both shapes @pchico83 ? (similar to how the build attribute takes a string, or a dict?)

@rberrelleza do you mean supporint:
```
sync:
  - .:/app
```
and?
```
sync:
  folders:
    - .:/app
  options:
    compression: true
```

This is doable. I would go with this one.

**Comment by rlamana** [comment]
+1!

File: pkg/syncthing/syncthing.go [hunk 1]
```diff
@@ -168,9 +169,9 @@ func New(dev *model.Dev) (*Syncthing, error) {
 		hash = []byte("")
 	}
 
-	rescanInterval := os.Getenv("OKTETO_RESCAN_INTERVAL")
-	if rescanInterval == "" {
-		rescanInterval = "300"
+	compression := "metadata"
+	if dev.Sync.Compression {
+		compression = "always"
```
Comment (rberrelleza) [inline pkg/syncthing/syncthing.go:174]: are there other potential options here? if so, maybe dev.Sync.Compression shouldn't be a bool but an enum?

File: pkg/model/dev.go [hunk 2]
```diff
@@ -462,6 +482,16 @@ func (dev *Dev) setDefaults() error {
 	}
 	dev.setRunAsUserDefaults(dev)
 
+	if os.Getenv("OKTETO_RESCAN_INTERVAL") != "" {
```
[the remark this answers is not in the document]
Comment (pchico83) [reply pkg/model/dev.go:485]: We can deprecate it, but let's keep it for a while first

File: pkg/syncthing/syncthing.go (hunk 1 above)
Comment (pchico83) [reply pkg/syncthing/syncthing.go:174]: There are other compression options, but I would keep it as a bool value for now to enable it or disable it

**Pull request merged**